src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs

```diff
@@ -354,9 +354,41 @@ private void SetClientPipeline(IChannel channel, Address remoteAddress)
                 var certificate = Settings.Ssl.Certificate;
                 var host = certificate.GetNameInfo(X509NameType.DnsName, false);
 
-                var tlsHandler = Settings.Ssl.SuppressValidation
-                    ? new TlsHandler(stream => new SslStream(stream, true, (_, _, _, _) => true), new ClientTlsSettings(host))
-                    : TlsHandler.Client(host, certificate);
+                IChannelHandler tlsHandler;
+
+                if (Settings.Ssl.SuppressValidation)
+                {
+                    // Test/dev mode: Accept any server certificate
+                    if (Settings.Ssl.RequireMutualAuthentication)
+                    {
+                        // Provide client cert for mutual TLS
+                        tlsHandler = new TlsHandler(
+                            stream => new SslStream(stream, true, (_, _, _, _) => true,
+                                (_, _, _, _, _) => certificate),
+                            new ClientTlsSettings(host));
+                    }
+                    else
+                    {
+                        // No client cert needed
+                        tlsHandler = new TlsHandler(
+                            stream => new SslStream(stream, true, (_, _, _, _) => true),
+                            new ClientTlsSettings(host));
+                    }
+                }
+                else
+                {
+                    // Production mode: Validate server certificate
+                    if (Settings.Ssl.RequireMutualAuthentication)
+                    {
+                        // Provide client cert for mutual TLS
+                        tlsHandler = TlsHandler.Client(host, certificate);
```

Comment:
I need to double check, behaviorally, how this compares to the suppressed validation scenario.

Reply:
Yep, this enforces mTLS when the certificate is provided. This is fine.